graf2d/graf/CMakeLists.txt

```diff
@@ -9,9 +9,9 @@
 # @author Pere Mato, CERN
 ############################################################################
 
-include_directories(${FREETYPE_INCLUDE_DIRS})
-
 ROOT_STANDARD_LIBRARY_PACKAGE(Graf
+  HEADERSDIR
+    graf2d/graf/inc 
```

Comment:
`ROOT_STANDARD_LIBRARY_PACKAGE` should know the component design of ROOT and should do that implicitly without using a `HEADERSDIR`.

The long-term advantage is that we pipe this information in a common place which makes introduction of new features easier. For example, if we decide to redesign the header locations we will not break user code presumably (think v7 headers).

We have discussed that in the context of the ROOT package manager: if we have non-standard locations (which cannot be deduced easily) we should have an non-public cmake 'component map' which holds this information. This would be helpful when we construct minimal (-Dminimal or alike) distributions of ROOT. We will just say: -Dcomponents=Graf and we should retrieve the needed information based on that map.

These set of changes seem quite close to this but making it explicit which allows less customization.

Reply:
`HEADERSDIR` is only for ROOT compilation. It says where headers of `Graf` library is located. Any library dependent on `Graf` have to use these directories in include paths.
But it is only for ROOT compilation. Any code outside ROOT should use `$ROOTSYS/includes` path.

Reply:
This should not be exposed with an argument. I think this should be internal to `ROOT_STANDARD_LIBRARY_PACKAGE`. We have all the information to deduce these locations.

Reply:
I fully remove this interface.
Now `standard` path like `${CMAKE_CURRENT_SOURCE_DIR}/inc` assigned as `PUBLIC`  via `target_include_directories` and can be discovered by other targets